File: 1417. Reformat The String/Solution.cpp
#include <cctype>
#include <queue>
#include <string>
using namespace std;

class Solution
{
public:
    string reformat(string s)
    {
        int n1 = 0, n2 = 0;     // n1 计数字母，n2计数数字
        for (const char &ch : s) {
            if (isalpha(ch))
                n1++;
            else
                n2++;
        }
        if (abs(n1 - n2) > 1)
            return "";
        if (n1 > n2) {
            n1 = 0;
            n2 = 1;
        }
        else {
            n1 = 1;
            n2 = 0;
        }
        string res;
        res.resize(s.size());
        for (const char &ch : s) {
            if (isalpha(ch)) {
                res[n1] = ch;
                n1 += 2;
            }
            else {
                res[n2] = ch;
                n2 += 2;
            }
        }
        return res;
    }
};

File: 1417. Reformat The String/Solution.py
class Solution:
    def reformat(self, s: str) -> str:
        s1 = [x for x in s if x.isalpha()]
        s2 = [x for x in s if x.isdigit()]
        if abs(len(s1) - len(s2)) > 1:
            return ''
        res = [None] * len(s)
        if len(s1) > len(s2):
            res[0::2] = s1
            res[1::2] = s2
        else:
            res[1::2] = s1
            res[0::2] = s2
        return ''.join(res)
